ast/src/analyzed/mod.rs

```diff
@@ -43,6 +44,7 @@ pub struct Analyzed<T> {
     pub source_order: Vec<StatementIdentifier>,
     /// Symbols from the core that were added automatically but will not be printed.
     pub auto_added_symbols: HashSet<String>,
+    pub implementations: HashMap<String, Vec<TraitImplementation<Expression>>>,
```

Comment:
Please use `trait_implementations` or `trait_impls`, I think just 'implementations' is too generic. Also please document.

Reply:
Oh and maybe move it up below definitions?